pkg/agent/qrm-plugins/network/dynamicpolicy/policy.go

```diff
@@ -0,0 +1,366 @@
+	"github.com/kubewharf/katalyst-core/pkg/util/general"
+	"github.com/kubewharf/katalyst-core/pkg/util/native"
+	"github.com/kubewharf/katalyst-core/pkg/util/qos"
+)
+
+const (
+	// NetworkResourcePluginPolicyNameDynamic is the policy name of dynamic network resource plugin
+	NetworkResourcePluginPolicyNameDynamic = "dynamic"
+	// ResourceNameNetwork is the resource name of network
+	ResourceNameNetwork = "network"
+)
+
+// DynamicPolicy is the dynamic network policy
+type DynamicPolicy struct {
+	sync.RWMutex
+
+	name       string
+	stopCh     chan struct{}
+	started    bool
+	qosConfig  *generic.QoSConfiguration
+	emitter    metrics.MetricEmitter
+	metaServer *metaserver.MetaServer
+
+	netClassMap                   map[string]uint32
+	isCgV2Env                     bool
+	applyNetClassFunc             func(podUID, containerId string, data *common.NetClsData) error
+	podLevelNetClassAnnoKey       string
+	podLevelNetAttributesAnnoKeys []string
+}
+
+// NewDynamicPolicy returns a dynamic network policy
+func NewDynamicPolicy(agentCtx *agent.GenericContext, conf *config.Configuration, _ interface{}, agentName string) (bool, agent.Component, error) {
+	wrappedEmitter := agentCtx.EmitterPool.GetDefaultMetricsEmitter().WithTags(agentName, metrics.MetricTag{
+		Key: util.QRMPluginPolicyTagName,
+		Val: NetworkResourcePluginPolicyNameDynamic,
+	})
+
+	policyImplement := &DynamicPolicy{
+		qosConfig:   conf.QoSConfiguration,
+		emitter:     wrappedEmitter,
+		metaServer:  agentCtx.MetaServer,
+		stopCh:      make(chan struct{}),
+		name:        fmt.Sprintf("%s_%s", agentName, NetworkResourcePluginPolicyNameDynamic),
+		netClassMap: make(map[string]uint32),
+	}
+
+	if common.IsCgroup2UnifiedMode() {
+		policyImplement.isCgV2Env = true
+		policyImplement.applyNetClassFunc = agentCtx.MetaServer.ExternalManager.ApplyNetClass
+	} else {
+		policyImplement.isCgV2Env = false
+		policyImplement.applyNetClassFunc = cgroupcmutils.ApplyNetClsForContainer
+	}
+
+	policyImplement.ApplyConfig(conf.DynamicConfiguration)
+
+	pluginWrapper, err := skeleton.NewRegistrationPluginWrapper(
+		policyImplement,
+		conf.QRMPluginSocketDirs, nil)
+	if err != nil {
+		return false, agent.ComponentStub{}, fmt.Errorf("dynamic policy new plugin wrapper failed with error: %v", err)
+	}
+
+	return true, &agent.PluginWrapper{GenericPlugin: pluginWrapper}, nil
+}
+
+// ApplyConfig applies config to DynamicPolicy
+func (p *DynamicPolicy) ApplyConfig(conf *config.DynamicConfiguration) {
+	p.Lock()
+	defer p.Unlock()
+
+	p.netClassMap[apiconsts.PodAnnotationQoSLevelReclaimedCores] = conf.NetClass.ReclaimedCores
+	p.netClassMap[apiconsts.PodAnnotationQoSLevelSharedCores] = conf.NetClass.SharedCores
+	p.netClassMap[apiconsts.PodAnnotationQoSLevelDedicatedCores] = conf.NetClass.DedicatedCores
+	p.netClassMap[apiconsts.PodAnnotationQoSLevelSystemCores] = conf.NetClass.SystemCores
+
+	p.podLevelNetClassAnnoKey = conf.PodLevelNetClassAnnoKey
```

Comment:
It seems that p.podLevelNetClassAnnoKey is not used?

Reply:
Yes. It has been fixed.